server/engine/src/main/java/org/eclipse/lsp/cobol/core/preprocessor/cbl/CblParser.java

```diff
@@ -0,0 +1,420 @@
+    one("SPACE");
+    int start = pos - 1;
+    one("(");
+    String next = next();
+    int i = checkIntegerLiteral(next);
+    if (i != 1 && i != 2 && i != 3) {
+      semanticError(next, "SPACE must be 1, 2 or 3.");
+    }
+    one(")");
+    directiveNodes.add(createDirectiveNode(start, pos));
+    removeSegments(start, pos);
+  }
+
+  private void parseLineCount() throws CblDiagnosticException {
+    or("LINECOUNT", "LC");
+    int start = pos - 1;
+    one("(");
+    String argument = next();
+    int i = checkIntegerLiteral(argument);
+    one(")");
+    directiveNodes.add(createDirectiveNode(start, pos));
+    removeSegments(start, pos);
+    if (isLineNumberWrong(i)) {
+      semanticError(argument, "LINECOUNT must be an integer between 1 and 255");
+    }
+  }
+
+  private void parseFlag() throws CblDiagnosticException {
+    or("FLAG", "F");
+    int start = pos - 1;
+    one("(");
+    or("E", "I", "S", "U", "W");
+    one(")");
+    directiveNodes.add(createDirectiveNode(start, pos));
+    removeSegments(start, pos);
+  }
+
+  private void removePrevComma(int from, int to) {
+    for (int i = to - 1; i >= from; i--) {
+      if (segments[i] == null || isBlank(segments[i])) {
+        continue;
+      }
+      if (",".equals(segments[i])) {
+        segments[i] = null;
+      }
+      return;
+    }
+  }
+
+  private void removeSegments(int start, int end) {
+    Arrays.fill(segments, start, end, null);
+  }
+
+  private CompilerDirectiveNode createDirectiveNode(int start, int end) {
+    StringBuilder text = new StringBuilder();
+    for (int i = start; i < end; i++) {
+      text.append(segments[i].trim());
+    }
+    Range range =
+        new Range(new Position(lineNumber, ranges[start]), new Position(lineNumber, ranges[end]));
+    Locality locality = Locality.builder().uri(uri).range(range).build();
+    return new CompilerDirectiveNode(locality, text.toString(), CICSDialect.DIALECT_NAME);
+  }
+
+  private void semanticError(String next, String message) {
+    diagnostics.add(
+        new CblDiagnosticException(makeCurrentLocation(), next, WARNING, message)
+            .toSyntaxError(lineNumber));
+  }
+
+  private boolean isLineNumberWrong(int nnn) {
+    // nnn must be an integer between 1 and 255.
+    if (nnn == 0) {
+      return true;
+    }
+    return nnn <= 0 || nnn > 255;
+  }
+
+  private int checkIntegerLiteral(String next) throws CblDiagnosticException {
+    try {
+      return Integer.parseInt(next);
+    } catch (NumberFormatException e) {
+      throw expect("integer", makeCurrentLocation(), "integer");
+    }
+  }
+
+  private Location makeCurrentLocation() {
+    Position start = new Position(lineNumber, ranges[pos - 1]);
+    Position end = new Position(lineNumber, ranges[pos]);
+    Range range = new Range(start, end);
+    return new Location(uri, range);
+  }
+
+  private void parseOption() throws CblDiagnosticException {
+    int depth = 0;
+    while (hasMore()) {
+      if (depth == 0)
```

Comment:
I think this code should also skip over quoted strings:
<img width="586" height="145" alt="image" src="https://github.com/user-attachments/assets/c4d1fba8-868d-4338-8b70-24d894f63d5d" />